spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -265,4 +284,28 @@ private static Schema validateOrMergeWriteSchema(
 
     return writeSchema;
   }
+
+  private boolean skipOrderingAndDistribution(OrderedDistribution distribution) {
+    // check if all input files have same partitioning as current table partitioning
+    boolean allInputFilesMatchTablePartitioning =
+        copyOnWriteScan.files().stream().allMatch(x -> x.file().specId() == table.spec().specId());
+
```

Comment:
Should we return immediately if it is a false (to save compute for other checks)?

Reply:
good point, updated